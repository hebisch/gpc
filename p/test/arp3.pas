program verRELset ( output ) ;
{gpc -fextended-pascal verRELset.pas}
type
  tMin  = 'a'..'z';
  tCMin = set of tMin;

const
  vowels = tCMin['a', 'e', 'i', 'o', 'u'];

var
  CM1, CM2 : tCMin;
  m        : tMin;
begin
   CM1 := ['a'..'r'] ;
   CM2 := ['j'..'l'] ;
   if CM1 < CM2 then writeln('CM1 < CM2'); {not standard}
   if CM1 > CM2 then writeln('CM1 > CM2'); {not standard}
   for m := 'a' to 'z' do
      if m in vowels then write ( m ); {right!}
{  for m in vowels do write ( m ); constant out of range }
end.
