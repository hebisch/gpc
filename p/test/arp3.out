CM1 > CM2
aeiou